include/oneapi/tbb/detail/_machine.h

```diff
@@ -132,7 +132,7 @@ constexpr std::uintptr_t number_of_bits() {
 
 // logarithm is the index of the most significant non-zero bit
 static inline uintptr_t machine_log2(uintptr_t x) {
-#if defined(__GNUC__) || defined(__clang__)
+#if (defined(__GNUC__) || defined(__clang__)) && !defined(__POWERPC__)
     // If P is a power of 2 and x<P, then (P-1)-x == (P-1) XOR x
     return (number_of_bits<decltype(x)>() - 1) ^ gnu_builtins::clz(x);
```

Comment:
according to clang [documentation](https://clang.llvm.org/docs/LanguageExtensions.html#builtin-functions) it supports GCC intrinsic (assuming [clz](https://github.com/oneapi-src/oneTBB/blob/797dfaa7fc81803ff4d5c39b47fb4adcc4a19252/include/oneapi/tbb/detail/_machine.h#L112)).

Reply:
Clang is not relevant for Darwin PPC case, but if GCC builtin is preferred over arch-specific implementation, let me try building with it. I will update on this soon.

Reply:
IMHO compiler intrinsic is easier to understand than platform specific asm code.

Reply:
Please keep in mind that we also have PowerPC on Linux and *BSD when using definitions like `__POWERPC__`.

If you mean "PowerPC on MacOS", you should test for `__APPLE__` as well.

Reply:
@glaubitz @anton-potapov Let me also check this and update too.